src/pytorch_lightning/strategies/fully_sharded_native.py

```diff
@@ -212,6 +212,12 @@ def _setup_model(self, model: torch.nn.Module) -> FullyShardedDataParallel:
             del self.kwargs["auto_wrap_policy"]
 
         log.detail(f"setting up FSDP model with device id: {self.root_device.index}, kwargs: {self.kwargs}")
+
+        rank_zero_info(
```

Comment:
@rohitgr7 
One last todo I have in my PR here #14967 is to validate that the optimizer is referencing the weights of the wrapped model. Once I have that code, I will let you know. We will have to do it here too. I believe it is better to only warn if necessary. In fact, we should error out, because if the optimizer is referencing the unwrapped model weights, the .step() will not update the right weights.

Reply:
> referencing the weights of the wrapped model.

I checked it, for the unwrapped model you get `ValueError: optimizer got an empty parameter list` so not sure if it's possible to detect it.

Reply:
Here is the code that I added in Lite FSDP (https://github.com/Lightning-AI/lightning/pull/14967):
Every optimizer that references parameters in a FSDP model must reference parameters of type FlatParameter. If that's not the case, the user must have messed up the order. 

 ```py
   def setup_optimizer(self, optimizer: Optimizer) -> Optimizer:
        """Set up an optimizer for a model wrapped with FSDP.
        This setup method doesn't modify the optimizer or wrap the optimizer. The only thing it currently does is
        verify that the optimizer was created after the model was wrapped with `:meth:setup_module` with a reference
        to the flattened parameters.
        """
        from torch.distributed.fsdp import FlatParameter

        num_groups = len(optimizer.param_groups)
        if num_groups > 1:
            raise ValueError(
                "An optimizer used with an FSDP model does not support multiple param groups."
                f" Found {num_groups} parameter groups."
            )

        if any(isinstance(param, FlatParameter) for param in optimizer.param_groups[0]["params"]):
            return optimizer

        raise ValueError(
            "The optimizer does not seem to reference any FSDP parameters. HINT: Make sure to create the optimizer"
            " after setting up the model."
        )
```

Reply:
If you want, we can make the validation code a utility function that can be shared between Lite and PL

Reply:
in lite if you initialize optimizer with old model (unwrapped), it doesn't raise any error?

Reply:
of course it will, that's the whole point of this message. see the test cases

Reply:
oh I see. I guess you are proposing an extension here to check whether optimizer is initialized with wrapped models. My motivation for the PR was to just let users know that since with unwrapped model an error is raised.

yes, we should use this common utility. We should do it for both native and fairscale. You can merge you PR first and then I'll continue here :)

qq: what if users use the unwrapped model and they get `ValueError: optimizer got an empty parameter list` with no clue what to do? Do you think we should keep this info message?